### ISSUE TITLE
Re-create authorization instances after refreshAccessToken() was called

### ISSUE THREAD

**Issue by Moongazer** [mautic/mautic-typo3#66]
**Is your feature request related to a problem? Please describe.**
We extending EXT:mautic-typo3 to submit API requests via EventListeners, which are called on certain events in our system (e.g. `UserBookedSomethingEvent`). If such an event is dispatched, we want to add a certain Tag to the related Mautic contact.

All this works very well, except the access-token stored in the Mautic YAML is expired! To make sure that we always have a valid token, the EventListener checks it before doing anything else:
```
    public function __invoke(UserBookedSomethingEvent $event): void
    {
        if ($authorizeService->accessTokenToBeRefreshed()) {
            $authorizeService->refreshAccessToken();
        }

    // we should be able to make API requests here, because the YAML config holds a valid access-token
```
But here comes the problem: yes, the YAML config is updated with a new access-token (in case the prior one was expired), but the singletons e.g. in `AbstractRepository->$authorization` and `AuthorizationFactory::$oAuth` still holding the old access-token! We're not able to set, overwrite or dynamically reload these instances, which makes the API request failing.

We was wondering why the singleton instances with the old YAML values are created so early, before the Mautic API is used through the repositories. The reason seems to be, that `AuthorizationFactory::createAuthorizationFromExtensionConfiguration()` is called already during the Container creation for DI very early in the TYPO3 life-circle, that's why the auth objects are created with the old values.

**Describe the solution you'd like**
Not sure what is a good solution for this problem, but here are some proposals:
1) Before returning the singleton `$oAuth` instance, `AuthorizationFactory::createAuthorizationFromExtensionConfiguration()` should always check if the YAML holds new values. If so, it should re-create the `$oAuth` instance and return it.
2) Adding a method `AbstractRepository->setAuthorization()` to manually overwrite the `OAuth` object during runtime. If then `AbstractRepository->getApi()` is called, it uses the new authorization object.

Do you have any other solutions for this problem? Discussion welcome. 

**Additional context**
TYPO3: v10.4.28
mautic-typo3: v4.1

**Comment by SZChimp** [comment]
I am not sure if this is exactly the same use case, but we just encountered the following issue:

We had a connected TYPO3 11.5 using a migrated mautic-typo3 extension. The authorization process worked fine until the refresh token expired after 14 days (which it does using the default configuration). The sys_registry still held the expired access token as well as the expired (and now invalid) refresh token. We had to manually delete the row from the sys_registry table to re-authorize the TYPO3 in mautic.
